File: src/main/java/com/github/tanokun/tanorpg/util/command/Command.java
package com.github.tanokun.tanorpg.util.command;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface Command {

    String parentName();

    String name();

    String desc();

    String flags() default "";
}

File: src/main/java/com/github/tanokun/tanorpg/util/command/Command.kt
package com.github.tanokun.tanorpg.util.command

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class Command(val parentName: String, val name: String, val desc: String, val flags: String = "")